noir-projects/aztec-nr/authwit/src/auth.nr

```diff
@@ -252,7 +252,7 @@ pub fn assert_inner_hash_valid_authwit(
  * @param on_behalf_of The address that have authorized the current call
  */
 // docs:start:assert_current_call_valid_authwit_public
-pub fn assert_current_call_valid_authwit_public(
+pub unconstrained fn assert_current_call_valid_authwit_public(
```

Comment:
Technically it is correct to say public code is "unconstrained" (meaning it is brillig), but we try to avoid the confusion to the user since the AVM is going to constrain it.

Would it be possible to use unsafe blocks in the oracles to avoid the abstraction leak?

Reply:
I tried but I can't return a slice to a constrained function :(